lib/core/include/metadata.hpp

```diff
@@ -0,0 +1,223 @@
+#ifndef METADATA_HPP
+#define METADATA_HPP
+
+#include <string>
+#include <vector>
+#include <map>
+
+#include "entity.hpp"
+#include "irods_query.hpp"
+
+#ifdef RODS_SERVER
+    #include "rsModAVUMetadata.hpp"
+    using rxComm_t = rsComm_t;
+    #define rxModAVUMetadata rsModAVUMetadata
+    #define IRODS_FILESYSTEM_ENABLE_SERVER_SIDE_API
+#else
+    #include "modAVUMetadata.h"
+    using rxComm_t = rcComm_t;
+    #define rxModAVUMetadata rcModAVUMetadata
+#endif
+
+#include "filesystem.hpp"
+#include "fmt/format.h"
+
+namespace irods::experimental::metadata {
+
+    struct avu {
+        std::string attribute;
+        std::string value;
+        std::string units;
+    };
+
+    auto operator==(const avu& _lhs, const avu& _rhs) {
+        bool eq = _lhs.attribute == _rhs.attribute &&
+                  _lhs.value     == _rhs.value &&
+                  _lhs.units     == _rhs.units;
+        return eq;
+    }
+
+    namespace tokens {
+        const std::string none{};
```

Comment:
`none` is likely not needed

Reply:
this mirrors the entity type enumeration in order to model the negative space.

Reply:
true, but looking at the impl, it appears that this value will never be encountered or used.  when might someone see this value?